RiotSwiftUI/Modules/Settings/ChangePassword/ChangePasswordModels.swift

```diff
@@ -0,0 +1,65 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+// MARK: View model
+
+enum ChangePasswordViewModelResult {
+    /// Submit with old and new passwords and sign out of all devices option
+    case submit(String, String, Bool)
+}
+
+// MARK: View
+
+struct ChangePasswordViewState: BindableState {
+    /// View state that can be bound to from SwiftUI.
+    var bindings: ChangePasswordBindings
+    
+    /// Whether the user can submit the form: old password should be entered, and new passwords should match
+    var canSubmit: Bool {
+        !bindings.oldPassword.isEmpty
+        && !bindings.newPassword1.isEmpty
```

Comment:
In the auth flow we make sure the password is 8 or more characters rather than non-empty. I wonder if its worth making some kind of password validation helper method somewhere where we could reuse this (and also add more complex requirements in the future to help match the [spec](https://spec.matrix.org/v1.2/client-server-api/#notes-on-password-management)).

Reply:
Although thinking about the last comment I made, this would be best if it was split into a length check and a validity check as we'd need to tell the user when the password was long enough, but had invalid characters. Gets complicated rather quickly 🙈

Reply:
It can get a bit complicated, but for example, an OptionSet with all requirements could pretty quickly adapt to any level of granularity desired for error message display.

Reply:
I've implemented a `PasswordValidator`.